### PR TITLE
Maps firebase connection

### LINKED ISSUE #48: refactor: get map activities from firebase
DoD

- [x] Activities (locations, titles, descriptions) are fetched from firebase instead of being hard-coded in the MapsActivity.

### PR DESCRIPTION
**Pull request by AthenaPK**
Closes #48

### CODE REVIEW

**Comment by aizensoosuke** [comment]
I can't really ask this from you, but please if you have time switch to compose, cause right now this is the only part of the project that still uses fragments and it's going to make switching to screens very annoying

File: app/src/main/java/com/github/studydistractor/sdp/procrastinationActivity/ProcrastinationActivity.kt [hunk 1]
```diff
@@ -5,15 +5,21 @@ import android.os.Parcelable
 
 data class ProcrastinationActivity(
     val name: String? = null,
+    val lat: Double? = null,
+    val long: Double? = null,
     val description: String? = null) :
     Parcelable {
     constructor(parcel: Parcel) : this(
         parcel.readString(),
+        parcel.readDouble(),
```
Comment (aizensoosuke) [inline app/src/main/java/com/github/studydistractor/sdp/procrastinationActivity/ProcrastinationActivity.kt:14]: aaaaaaaargh parcel :(

**Review by aizensoosuke — DISMISSED** [review]
Looks good to me, nice :)

File: app/src/main/java/com/github/studydistractor/sdp/procrastinationActivity/ProcrastinationActivity.kt (hunk 1 above)
Comment (AthenaPK) [reply app/src/main/java/com/github/studydistractor/sdp/procrastinationActivity/ProcrastinationActivity.kt:21]: Sorry, we can refactor later : P

File: app/src/androidTest/java/com/github/studydistractor/sdp/sillysillycoverage/SillySillyCoverageTest.kt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        assertEquals(16.sp, Typography.bodyLarge.fontSize)
+    }
+
+    @Test
+    fun testFirebaseLoginAuth() {
+        var auth = FirebaseLoginAuth(Firebase.auth)
+        auth.loginWithEmail("pouet", "pouet")
+        // no assertion, this is purely for artificial raising of the coverage
+    }
+}
```
Comment (abouquet27) [inline app/src/androidTest/java/com/github/studydistractor/sdp/sillysillycoverage/SillySillyCoverageTest.kt:75]: This should be avoided

File: app/src/main/java/com/github/studydistractor/sdp/maps/MapsActivity.kt [hunk 3]
```diff
@@ -63,27 +64,47 @@ class MapsActivity : AppCompatActivity(), OnMapReadyCallback {
...
     }
 
+    /**
+     * Displays the places on the map and launches the distraction activity when clicking on the marker
+     */
+    private fun displayPlaces() {
+        FireBaseProcrastinationActivityService().fetchProcrastinationActivities { activities ->
+            for (activity in activities) {
+                displayActivity(activity)
+            }
```
Comment (abouquet27) [inline app/src/main/java/com/github/studydistractor/sdp/maps/MapsActivity.kt:100]: The names are misleading I think. The function is called displayPlaces but it uses displayActivity

File: app/src/main/java/com/github/studydistractor/sdp/ui/LoginScreen.kt [hunk 4]
```diff
@@ -1,23 +1,129 @@
...
+    ) {
+        val email = remember { mutableStateOf(TextFieldValue("")) }
+        val password = remember { mutableStateOf(TextFieldValue("")) }
+        Text(
+            text = "Log In",
+            style = MaterialTheme.typography.titleLarge,
+            modifier = Modifier.padding(vertical = 32.dp)
+        )
+
+        OutlinedTextField(
```
Comment (abouquet27) [inline app/src/main/java/com/github/studydistractor/sdp/ui/LoginScreen.kt:52]: Need more description #65

File: app/src/main/java/com/github/studydistractor/sdp/ui/LoginScreen.kt [hunk 5]
```diff
@@ -1,23 +1,129 @@
...
+            ),
+            leadingIcon = {
+                Icon(Icons.Filled.Email, contentDescription = null)
+            },
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(vertical = 8.dp)
+                .testTag("email")
+        )
+        OutlinedTextField(
```
Comment (abouquet27) [inline app/src/main/java/com/github/studydistractor/sdp/ui/LoginScreen.kt:69]: Need more description #65

**Review by abouquet27 — DISMISSED** [review]
Mostly the code is well done. For the UI part of the code, it lacks some description which will be done in the next sprint according to issue #65

File: app/src/androidTest/java/com/github/studydistractor/sdp/sillysillycoverage/SillySillyCoverageTest.kt (hunk 2 above)
Comment (AthenaPK) [reply app/src/androidTest/java/com/github/studydistractor/sdp/sillysillycoverage/SillySillyCoverageTest.kt:75]: I agree, it was already on main though (not part of my commits).

File: app/src/main/java/com/github/studydistractor/sdp/ui/LoginScreen.kt (hunk 4 above)
Comment (AthenaPK) [reply app/src/main/java/com/github/studydistractor/sdp/ui/LoginScreen.kt:52]: Ok, let's take care of it during refactoring.

File: app/src/main/java/com/github/studydistractor/sdp/ui/LoginScreen.kt (hunk 5 above)
Comment (AthenaPK) [reply app/src/main/java/com/github/studydistractor/sdp/ui/LoginScreen.kt:69]: Ok, let's take care of it during refactoring.

**Review by bluedrack — APPROVED** [review]
LGTM.

**Pull request merged**